hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/TestCapacitySchedulerAsyncScheduling.java

```diff
@@ -927,7 +927,11 @@ public void testReleaseOutdatedReservedContainer() throws Exception {
      * First proposal should be accepted, second proposal should be rejected
      * because it try to release an outdated reserved container
      */
-    MockRM rm1 = new MockRM();
+    // disable async-scheduling for simulating complex scene
+    Configuration disableAsyncConf = new Configuration(conf);
+    disableAsyncConf.setBoolean(
+        CapacitySchedulerConfiguration.SCHEDULE_ASYNCHRONOUSLY_ENABLE, false);
```

Comment:
Why disable async scheduler in UT? And enable by default? I think we'd better enable async schedule by default.

Reply:
The following are the reasons why async scheduling is disabled in UT :

1. unlike async scheduling, sync scheduling gives more flexibility while writing UTs to mimic complex scenarios since the UT writer will have full control over when to schedule, where as with async scheduling this cannot be done.

2. There were 200+ failing when async scheduling was enabled (due to scheduling assertion)

Reply:
ok,get it.